### ISSUE TITLE
Invalid JIRA issue 'GLW-000' allowed through

### ISSUE THREAD

**Issue by davidholsgrove** [mohamicorp/yet-another-commit-checker#87]
A 'clever' developer has found a loophole where including 'GLW-000' in his commit message allows it get past the valid JIRA issue check.

I have 'Require Valid JIRA Issue' enabled, 'Ignore Unknown Keys' disabled, and a Issue JQL Matcher set to;

(issuetype = "Development sub-task" and resolution is EMPTY) 

Can this be blocked? :)

**Comment by sford** [comment]
Hey @davidholsgrove , is 'GLW-000' the only JIRA issue present in the commit message? And, is that a literal issue '000'? As in, YACC allows this even though an issue with number '000' shouldn't exist?

**Comment by davidholsgrove** [comment]
Yeah, its a simple commit message - one line, with 'GLW-000' at the start of line;

![capture](https://cloud.githubusercontent.com/assets/4031547/10468282/9647b41e-7241-11e5-8bb2-749a5175057e.PNG)

GLW-000 doesnt exist in Jira, the first valid ticket would be GLW-001 (which in Jira resolves to GLW-1)

![capture2](https://cloud.githubusercontent.com/assets/4031547/10468281/96475a78-7241-11e5-9b86-8d7aa6546daf.PNG)

thanks (love the addon btw!)

**Comment by sford** [comment]
Hey @davidholsgrove , I just tried to reproduce this error using local dev Stash and JIRA (using `atlas-run` from Atlassian SDK) and am having some trouble. I created a TEST project and tried committing `TEST-000`, and it was blocked:

    remote: refs/heads/master: 92a0e11f2c6a68089a261fe8f0b91ffbf0638030: TEST-000: JIRA Issue does not exist

Could there be another YACC setting that might be causing this commit to be allowed? Like `Exclude by Regex`? Or even allow service users if the the developer is using an SSH access key for system user (probably unlikely, but possible)?
